todo_app/data/session_items.py

```diff
@@ -67,3 +67,12 @@ def save_item(item):
     session['items'] = updated_items
 
     return item
+
+def delete_item(id):
+    existing_items = get_items()
+    try:
+        existing_items.pop(id)
+    finally:
+        updated_items = existing_items
+        session['items'] = updated_items
```

Comment:
The finally block doesn't really belong inside the "finally" - isn't it meaningless to update session['items'] if `existing_items.pop` threw an error?

And the two lines 
```
        updated_items = existing_items
        session['items'] = updated_items
```
are exactly equivalent to `session['items'] = existing_items`